src/OrchardCore.Modules/OrchardCore.Contents/Controllers/ApiController.cs

```diff
@@ -67,7 +67,7 @@ public async Task<IActionResult> Delete(string contentItemId)
                 return this.ChallengeOrForbid("Api");
             }
 
-            var contentItem = await _contentManager.GetAsync(contentItemId);
+            var contentItem = await _contentManager.GetAsync(contentItemId, VersionOptions.Latest);
```

Comment:
The current method will remove only a ContentItem if it is published.
But this would allow removing also drafts.

Reply:
Yes, I tried to delete content through the API and always returned 204 status

Reply:
I changed the code to make it consistent with the Action code on the MVC page
https://github.com/OrchardCMS/OrchardCore/blob/d6afd678791f399aeca23765a943f0702f2c9802/src/OrchardCore.Modules/OrchardCore.Contents/Controllers/AdminController.cs#L661